django_filters/rest_framework/backends.py

```diff
@@ -1,30 +1,40 @@
+from __future__ import annotations
+
 import warnings
+from typing import TYPE_CHECKING, Any, TypeVar
 
 from django.template import loader
 
 from .. import compat, utils
 from . import filters, filterset
 
+if TYPE_CHECKING:
+    from django.db.models.query import QuerySet
+    from rest_framework.generics import GenericAPIView
+    from rest_framework.request import Request
+
+    Q = TypeVar("Q", bound=QuerySet[Any])
+
 
 class DjangoFilterBackend:
     filterset_base = filterset.FilterSet
     raise_exception = True
 
     @property
-    def template(self):
+    def template(self) -> str:
         if compat.is_crispy():
             return "django_filters/rest_framework/crispy_form.html"
         return "django_filters/rest_framework/form.html"
 
-    def get_filterset(self, request, queryset, view):
+    def get_filterset(self, request: Request, queryset: QuerySet[Any], view: GenericAPIView) -> filterset.FilterSet | None:
         filterset_class = self.get_filterset_class(view, queryset)
         if filterset_class is None:
             return None
 
         kwargs = self.get_filterset_kwargs(request, queryset, view)
         return filterset_class(**kwargs)
 
-    def get_filterset_class(self, view, queryset=None):
+    def get_filterset_class(self, view: GenericAPIView, queryset: QuerySet[Any] | None = None) -> type[filterset.FilterSet] | None:
```

Comment:
If I'm honest, I just don't want this in my code. `queryset: QuerySet[Any]` — like really? 

(sigh)

Reply:
Well, it's the best we have for now, and it's very useful.

Actually, there should be `QuerySet[_MT]`, where `_MT = TypeVar("_MT", bound=Model)`, and it allows type-checker and LSP server to know that when you access `queryset.model._meta` - it's model options, and your editor can suggest all relevant attributes when you need it.

Reply:
@last-partizan Yes, I understand how it works. I just don't like looking at it, or writing it.